### PR TITLE
#6Task. Nikita Sharshun

### PR DESCRIPTION
**Pull request by sharshoon**
https://travis-ci.org/github/sharshoon/js-assignments/builds/706145405

### CODE REVIEW

File: task/06-conditions-n-loops-tasks.js [hunk 1]
```diff
@@ -30,10 +29,18 @@
  *
  */
 function getFizzBuzz(num) {
-    throw new Error('Not implemented');
+  switch (true) {
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: makes no sense. Why do you use switch here?

File: task/06-conditions-n-loops-tasks.js [hunk 2]
```diff
@@ -63,10 +73,11 @@ function getFactorial(n) {
  *   -1,1  =>  0  ( = -1 + 0 + 1 )
  */
 function getSumBetweenNumbers(n1, n2) {
-    throw new Error('Not implemented');
+  let result = 0;
+  while (n2 != n1 - 1) result += n2--;
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: do not modify arguments if not required. Do not do assignements in the loop definition

File: task/06-conditions-n-loops-tasks.js [hunk 3]
```diff
@@ -81,79 +92,82 @@ function getSumBetweenNumbers(n1, n2) {
...
  *     width: 20,
  *     height: 10
  *  }
- * 
+ *
  *  (5;5)
- *     -------------  
- *     |           | 
+ *     -------------
+ *     |           |
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: do not modify formattting of the comments

File: task/06-conditions-n-loops-tasks.js [hunk 4]
```diff
@@ -81,79 +92,82 @@ function getSumBetweenNumbers(n1, n2) {
...
+ *       x: 5,
  *       y: 5
- *     },        
+ *     },
  *     radius: 20
  *  }
- * 
- * Point is object of 
+ *
+ * Point is object of
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: do not modify formatting of the comments

File: task/06-conditions-n-loops-tasks.js [hunk 5]
```diff
@@ -251,9 +271,23 @@ function reverseInteger(num) {
  *   4916123456789012 => false
  */
 function isCreditCardNumber(ccn) {
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js:284]: indentations are not consisten in this method

File: task/06-conditions-n-loops-tasks.js [hunk 6]
```diff
@@ -270,10 +304,24 @@ function isCreditCardNumber(ccn) {
...
+  function getSum(n){
+    let res = 0
+    while(n != 0){
+      res += n % 10
+      n = Math.floor(n / 10)
+    }
+    return res
+  }
+
+  let result = 0
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: missing semilcolon

File: task/06-conditions-n-loops-tasks.js [hunk 7]
```diff
@@ -332,10 +392,37 @@ function isBracketsBalanced(str) {
  *
  */
 function timespanToHumanString(startDate, endDate) {
-    throw new Error('Not implemented');
+  function roundDate(num){
+    return num % 1 > 0.5 ? Math.round(num) : Math.floor(num)
+  }
+  
+  let seconds = (endDate - startDate) / 1000
+  switch(true){
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: refactore this, using switch(true) makes no sense

File: task/06-conditions-n-loops-tasks.js [hunk 8]
```diff
@@ -356,10 +443,14 @@ function timespanToHumanString(startDate, endDate) {
  *    365, 10 => '365'
  */
 function toNaryString(num, n) {
-    throw new Error('Not implemented');
+  let result = ""
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: indentation

File: task/06-conditions-n-loops-tasks.js [hunk 9]
```diff
@@ -431,27 +542,72 @@ function getMatrixProduct(m1, m2) {
...
-}
 
+  for(let i = 0; i < position.length; i++){
+    let count = 1
+
+    for(let j = 1; j < position.length; j++){
+      if (position[i][j-1] != position[i][j] || !position [i][j])
+        break
+      else 
+        count++
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: brackets

File: task/06-conditions-n-loops-tasks.js [hunk 10]
```diff
@@ -431,27 +542,72 @@ function getMatrixProduct(m1, m2) {
...
+  getFactorial: getFactorial,
+  getSumBetweenNumbers: getSumBetweenNumbers,
+  isTriangle: isTriangle,
+  doRectanglesOverlap: doRectanglesOverlap,
+  isInsideCircle: isInsideCircle,
+  findFirstSingleChar: findFirstSingleChar,
+  getIntervalString: getIntervalString,
+  reverseString: reverseString,
+  reverseInteger: reverseInteger,
+  isCreditCardNumber: isCreditCardNumber,
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: keep original formatting

**Review by AlexMokin — COMMENTED** [review]
- Review this https://javascript.info/coding-style and fix your code accoringly to this code style. 
- use 4 spaces for indetntations
- lots of missing semicolons
- lots of missing brackets

I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

File: task/06-conditions-n-loops-tasks.js [hunk 11]
```diff
@@ -332,10 +391,45 @@ function isBracketsBalanced(str) {
...
-    throw new Error('Not implemented');
+    function roundDate(num) {
+        return num % 1 > 0.5 ? Math.round(num) : Math.floor(num);
+    }
+
+    let seconds = (endDate - startDate) / 1000;
+    if(seconds <= 45){
+        return "a few seconds ago";
+    }
+    else if(seconds <= 90){
```
Comment (AlexMokin) [inline task/06-conditions-n-loops-tasks.js]: else can be omitted if you return in the if above

**Review by AlexMokin — COMMENTED** [review]
Looks better. But I still see changes to the original formatting of the comments, e.g. see image below, it's not only this place, see changes for the whole file and make sure formatting of the comments is not changed. Please fix the issue with comments, the rest is alright.
image

I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by sharshoon** [comment]
I copied the tasks from master branch and pasted all the tasks-solutions here once again. So it should be right!

**Comment by AlexMokin** [comment]
accepted

**Pull request closed without merging**